aodndata/acorn/handler.py

```diff
@@ -34,31 +34,49 @@ def get_fields_from_filename(filename):
 
 def get_type(filepath):
     """return acorn_file_type, the file type of an ACORN file based on its filename"""
+    file_basename = os.path.basename(filepath)
+
+    if ACORN_FILE_PATTERN.match(file_basename):
+        fields = get_fields_from_filename(file_basename, pattern=ACORN_FILE_PATTERN)
+        product_type = fields['product_type']
+        file_version = fields['file_version']
+        platform_code = fields['platform_code']
+
+        if product_type == 'radial' and file_version == 'FV00':
+            acorn_file_type = "radial"
+
+        elif product_type == 'radial' and file_version == 'FV01':
+            acorn_file_type = "radial_quality_controlled"
+
+        elif product_type == 'sea-state' and file_version == 'FV00':
+            acorn_file_type = "vector"
+
+        elif product_type == 'wavespec' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-wave-spectra_QC"
+
+        elif product_type == 'windp' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-wind-map_QC"
+
+        elif product_type == 'wavep' and file_version == 'FV01':
+            site_map_station = ['CBG', 'SAG', 'ROT', 'COF']
+
+            if any(s == platform_code for s in site_map_station):
+                acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
+            else:
+                acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV00':
+            acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-current-map_QC"
 
-    if "FV00_radial.nc" in filepath:
-        acorn_file_type = "radial"
-    elif "FV00_sea-state.nc" in filepath:
-        acorn_file_type = "vector"
-    elif "FV01_radial.nc" in filepath:
-        acorn_file_type = "radial_quality_controlled"
-    elif "FV01_wavespec.nc" in filepath:
-        acorn_file_type = "gridded_1h-avg-wave-spectra_QC"
-    elif "FV01_windp.nc" in filepath:
-        acorn_file_type = "gridded_1h-avg-wind-map_QC"
-    elif "FV01_wavep.nc" in filepath:
-        site_map_station = ('_CBG_', '_SAG_', '_ROT_', '_COF_')
-        if any(s in filepath for s in site_map_station):
-            acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
         else:
-            acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
-    elif "FV00_1-hour-avg.nc" in filepath:
-        acorn_file_type = "gridded_1h-avg-current-map_non-QC"
-    elif "FV01_1-hour-avg.nc" in filepath:
-        acorn_file_type = "gridded_1h-avg-current-map_QC"
+            raise InvalidFileNameError("file name: \"{filename}\" Unknown product type from filename".
+                                       format(filename=os.path.basename(filepath)))
```

Comment:
Could have this line present only once in your function (instead of being repeated twice). Error would be raised if acorn_file_type hasn't been set for example.

Reply:
definitely